doc/security/specs/device_provisioning/index.md

```diff
@@ -0,0 +1,977 @@
+    devices have secure boot enabled, which provides a logical locking mechanism
+    to restrict the use of open samples.
+8.  The Silicon Creator may issue a Certificate Revocation List (CRL) to the
+    device registry. The registry is in charge of serving the CRL to downstream
+    consumers (Silicon Owners).
+
+### Secure Boot
+
+Secure boot is always enforced by the mask ROM and cannot be disabled.
+Personalization and factory software payloads are signed, and boot verification
+is used to anchor the mechanism in which the device authenticates the
+provisioning appliance during personalization.
+
+### Injection Process
+
+This section describes the personalization injection mode in detail.
+
+<img src="img_provisioning_injection.svg" style="width: 900px;">
+
+**P0. Get device identifier**
+
+The DEVICE is initially in LOCKED state before the start of production test. The
+TESTER gets the Device ID via TAP interface. The provisioning appliance returns
+unlock tokens associated with the Device ID.
+
+Note 1: The Device ID read requirement applies only if unlock tokens are unique
+per device.
+
+Note 2: The manufacturer must document the unlock token usage plan, including
+the proposed rotation plan.
+
+**P1. Unlock device and load personalization software**
+
+P1.1. The TESTER unlocks the DEVICE by sending the `TEST_UNLOCK` and `TEST_EXIT`
+tokens via TAP interface.
+
+P1.2. The TESTER loads personalization software on the DEVICE using the SPI
+bootstrap interface followed by DEVICE reset. The DEVICE verifies the
+personalization software signature against one of the public keys stored in the
+ROM via secure boot.
+
+**P2. Device authentication and key exchange**
+
+P2.1 Authentication function
+
+The authentication function relies on the following assumptions:
+
+*   The provisioning appliance (HSM) is the only component trusted with secrets
+    other than the Device Under Test (DUT) in the manufacturing floor.
+*   The authentication function can only be implemented by trusted components.
+*   The wafer manufacturer does not collude with the OSAT to attack the
+    authentication function, although some mitigations are considered.
+
+```
+# Shared secret key between provisioning appliance and the DEVICE.
+# The parameters of the authentication function are outside the
+# classification scope of this document.
+key_auth = authentication_function()
+```
+
+P2.2. DEVICE sends authentication data to TESTER
+
+**The DEVICE will fail to generate an authentication payload if the device is
+not in `PROD`, `PROD_END` or `DEV` state.**
+
+The DEVICE sends an authentication payload to the TESTER including:
+
+*   Device ID (`device_id`).
+*   Static key (`receiver_pub_key`) used to derive session keys[^1]; and,
+*   MAC tag covering the authentication payload.
+
+The DEVICE sends the payload to the TESTER via SPI, repeating the message
+continuously to simplify timing adjustments in the test sequence.
```

Comment:
Yes, this is to try to simplify the synchronization between the tester and the DUT since the tester does not have a nice programing interface (it is just a binary vector).